content/advent-2017/tracking-the-stars.md

```diff
@@ -0,0 +1,286 @@
++++
+author = ["Scott Mansfield "]
+date = 2017-12-11T06:40:42Z
+title = "Tracking the Stars"
+series = ["Advent 2017"]
++++
+
+Back in April this year (2017) I thought it would be fun to try setting up a system to track the
+star counts of the top 1000 Go repositories on GitHub. This article describes how I did this data
+collection and some simple analysis of the dataset.
+
+I want to be clear that this was for fun only and I'm not advocating that the number of stars a
+repository has is the be-all-end-all of its success. There's many, many repositories I am not
+mentioning here that are high quality, useful code. There's also tons of code that lives in private
+repositories outside Github or on Bitbucket and GitLab. In the context of this article, the stars on
+GitHub are a set of data used to explore the patterns people have when starring repositories.
+
+## Collecting the data
+
+The collection of the stars could be an article all its own (and may become one elsewhere) so I will
+be brief.
+
+The collection is done by a [Lambda function](/postimages/advent-2017/tracking-the-stars/lambda.py)
+on Amazon Web Services using a timer that triggers an execution every 15 minutes. The lambda
+function collects the stars count of the top 1000 Go repositories, which is the maximum the GitHub
+search API will return. This is a post about Go, so the lambda function is written in... Python, of
+course! The total resource usage is well below the free tier limits, so running this function is
+free indefinitely.
+
+The data that the lambda function collects is stored in a DynamoDB table that just holds the repo
+name, an integer timestamp (epoch time in seconds), and the number of stars at that time. By the
+time this article was started, I had several hundred megabytes of data in DynamoDB. As with the
+Lambda function, the usage here is below the free tier limit, so the storage is free as well. For
+reference, the allocated read and write capacity is 20 units each.
+
+Just getting the data out of DynamoDB was a chore, as I chose to use the Data Pipeline in AWS which
```

Comment:
I'd try to avoid phrases like "X was a chore" or "that's fairly easy to figure out".  I think that the tone should try to remain neutral and not make any assumptions about prior reader knowledge.

Reply:
Good point. Rephrasing